build.gradle

```diff
@@ -10,6 +11,13 @@ buildscript {
 
     dependencies {
         classpath(libs.kotlinx.teamInfraGradlePlugin)
+
+        String kotlinVersion = providers.gradleProperty("kotlin_version").orNull
+        if (kotlinVersion != null && !kotlinVersion.isBlank()) {
+            constraints {
+                classpath("org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlinVersion")
```

Comment:
Should we do something like that instead: https://docs.gradle.org/current/userguide/platforms.html#sec:overwriting-catalog-versions?
To ensure that the proper version will be propagated everywhere (here, for example: https://github.com/Kotlin/kotlinx-benchmark/blob/0764551f1dde18955f986fbf1dd581449202a23f/integration/build.gradle.kts#L27)

Reply:
Very good point. Probably yes, it makes sense to override the version to make sure it's aligned. I wanted to avoid it because it will affect the KGP used to build the Gradle plugin, but actually, now that I think about it, the Gradle plugin is an included build and so it shouldn't be affected...

I'll try fixing it.

Reply:
Done, version catalog is overridden in `settings.gradle`, please take a look.

Reply:
Thanks! Do we still need this particular change with the `constraint` if now the version in the catalog will be overridden by a `kotlin_version` value?

Reply:
Hmmm practically we could remove the constraint, but Gradle dependency resolution can be  is weird and confusing and I think it would be easy to accidentally modify the resolved version. The assertions would help catch this, but I'd feel more confident keeping the constraint.

Reply:
Ok. I don't have a particular opinion about the assertion, perhaps @dkrasnoff could add something about it.

Otherwise, the change looks good to me.

Reply:
To me the only place looks more convenient to support.
If we have two places we should somewhere comment on why we have these two places, just help "future us" to remember :)

Reply:
I'd be happy to document it, but I'm not sure where...

Maybe CONTRIBUTING.md? But it doesn't exist yet, despite a link in the README...

https://github.com/Kotlin/kotlinx-benchmark/blob/0764551f1dde18955f986fbf1dd581449202a23f/README.md?plain=1#L435

Reply:
What about leaving a comment right here, where you add constraints with an explanation?

Reply:
Good idea, done. I also added a brief explanation to the Version Catalog override.